plugin-api.d.ts

```diff
@@ -899,6 +904,9 @@ interface DocumentNode extends BaseNodeMixin {
    * to call node.children.find(callback) or node.findChild(callback)
    */
   findOne(callback: (node: PageNode | SceneNode) => boolean): PageNode | SceneNode | null
+
+  findAllWithCriteria(criteria: { types: NodeType[] }): Array<PageNode | SceneNode>
+  findAllByType<T extends keyof NodeTypeMap>(type: T): NodeTypeMap[T]
```

Comment:
Would something like this work?

`findAllByType<T: NodeType>(type: T): { type: T } & (PageNode | SceneNode)`

or
```
SpecificNodeType<T> extends BaseNode {
  type: T
}
findAllByType<T: NodeType>(type: T): SpecificNodeType<T>
```

I feel like something like this should be doable, but I'm not sure. What you have is totally fine too, we just need to remember to add to the map whenever we add a new type.

Reply:
🙏 Yes, you're right. This is much cleaner! https://github.com/figma/plugin-typings/pull/104/commits/b78883ddbd72da9df7f189fe27879d8c73519ef7